server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -872,6 +867,16 @@ public VolumeVO allocVolume(CreateVolumeCmd cmd) throws ResourceAllocationExcept
                 _uuidMgr.generateUuid(Volume.class, cmd.getCustomId()), details);
     }
 
+    @Override
+    public void validateCustomDiskOfferingSizeRange(Long sizeInGB) {
+        Long customDiskOfferingMaxSize = VolumeOrchestrationService.CustomDiskOfferingMaxSize.value();
+        Long customDiskOfferingMinSize = VolumeOrchestrationService.CustomDiskOfferingMinSize.value();
+
+        if ((sizeInGB < customDiskOfferingMinSize) || (sizeInGB > customDiskOfferingMaxSize)) {
+            throw new InvalidParameterValueException("Volume size: " + sizeInGB + "GB is out of allowed range. Max: " + customDiskOfferingMaxSize + " Min:" + customDiskOfferingMinSize);
```

Comment:
```suggestion
            throw new InvalidParameterValueException(String.format("Volume size:  %sGB is out of allowed range. Min:  %s. Max: %s." sizeInGB, customDiskOfferingMinSize, customDiskOfferingMaxSize));
```

Reply:
@weizhouapache , will you apply this suggestion?

Reply:
@DaanHoogland done